contracts/erc20/erc20_capped.aes

```diff
@@ -0,0 +1,103 @@
+contract ERC20Capped =
+  record state = {
+    _owner            : address,
+    _cap              : int,
+    _totalSupply      : int,
+    _balances         : map(address, int),
+    _allowed          : map((address,address), int)}
+
+  public stateful function init(cap : int) : state = 
+    require(cap > 0, "Cap can't be less than a zero")
+    { _owner       = Call.caller,
+      _cap         = cap,
+      _totalSupply = 0,
+      _balances    = {},
+      _allowed     = {}}
+
+  private function lookupByAddress(k : address, m, v) =
+  	switch(Map.lookup(k, m))
+	    None    => v
+	    Some(x) => x
+
+  public function totalSupply() : int = state._totalSupply
```

Comment:
Maybe add function to get/show the cap ?

Reply:
Done